### PR TITLE
tools: logger: Add LDC dump option

### PR DESCRIPTION
**Pull request by ktrzcinx**
Create dump options for logger to print important information from
LDC file, like DBG_ABI version and UUID with corresponding names.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

Output:

```
$./sof-logger -i log/dma_trace_1.bin -l log/sof-cnl.ldc -d
logger ABI Version is   4:2:0
ldc_file ABI Version is 4:2:0

Components uuid dictionary size:        824 bytes
Components uuid base address:           0x1FFFA000
Components uuid entries:
           ADDRESS                                    UUID NAME
        0x1FFFA000  <8b9d100c-6d78-418f-90a3-e0e805d0852b> host
        0x1FFFA01C  <f11818eb-e92e-4082-82a3-dc54c604ebb3> pipe-task
        0x1FFFA03C  <34dc0385-fc2f-4f7f-82d2-6cee444533e0> volume-task
        0x1FFFA05C  <b77e677e-5ff4-4188-af14-fba8bdbf8682> volume
        0x1FFFA078  <c1c5326d-8390-46b4-aa47-95c3beca6550> src
        0x1FFFA090  <43a90ce7-f3a5-41df-ac06-ba98651ae6a3> eq-fir
        0x1FFFA0AC  <5150c0e6-27f9-4ec8-8351-c705b642d12f> eq-iir
        0x1FFFA0C8  <04e3f894-2c5c-4f2e-8dc1-694eeaab53fa> tone
        0x1FFFA0E4  <bc06c037-12aa-417c-9a97-89282e321a76> mixer
        0x1FFFA100  <c4b26868-1430-470e-a089-15d1c77f851a> mux
        0x1FFFA118  <385cc44b-f34e-4b9b-8be0-535c5f43a825> switch
        0x1FFFA134  <c2b00d27-ffbc-4150-a51a-245c79c5e54b> dai
        0x1FFFA14C  <e50057a5-8b27-4db4-bd79-9a639cee5f50> kpb-task
        0x1FFFA16C  <d8218443-5ff3-4a4c-b388-6cfe07b9562e> kpb
        0x1FFFA184  <55a88ed5-3d18-46ca-88f1-0ee6eae9930f> selector
        0x1FFFA1A4  <eba8d51f-7827-47b5-82ee-de6e7743af67> kd-test
        0x1FFFA1C0  <c8ec72f6-8526-4faf-9d39-a23d0b541de2> asrc
        0x1FFFA1DC  <bc9ebe20-4577-41bb-9eed-d0cb236328da> hda-dai
        0x1FFFA1F8  <8fa1d42f-bc6f-464b-867f-547af08834da> ipc-task
        0x1FFFA218  <59c87728-d8f9-42f6-b89d-5870a87b0e1e> dmic-work
        0x1FFFA238  <aafc26fe-3b8d-498d-8bd6-248fc72efa31> dmic-dai
        0x1FFFA258  <31458125-95c4-4085-8f3f-497434cb2daf> ssp-dai
        0x1FFFA274  <a8e4218c-e863-4c93-84e7-5c27d2504501> alh-dai
        0x1FFFA290  <b90f5a4e-5537-4375-a1df-95485472ff9e> comp-task
        0x1FFFA2B0  <a5dacb0e-88dc-415c-a1b5-3e8df77f1976> idc-cmd-task
        0x1FFFA2D4  <c63c4e75-8f61-4420-9319-1395932efa9e> agent-work
        0x1FFFA2F4  <37f1d41f-252d-448d-b9c4-1e2bee8e1bf1> main-task
        0x1FFFA314  <2b972272-c5b1-4b7e-926f-0fc5cb4c4690> dma-trace-task
        -------------------------------------------------- cnt: 28
```

### CODE REVIEW

File: tools/logger/convert.c [hunk 1]
```diff
@@ -80,6 +80,32 @@ static void log_err(FILE *out_fd, const char *fmt, ...)
...
+char *format_uid_raw(const struct sof_uuid *uid_val, int use_colors,
+		     int name_first)
+{
+	const char *name = (const char *)(uid_val + 1) + 4;
+	int off;
+	char *str;
+
+	str = calloc(1, (use_colors ? strlen(KBLU) : 0) + 1 +
+		36 + 1 + *(uint32_t *)(uid_val + 1) + 1 +
+		(use_colors ? strlen(KNRM) : 0));
```
Comment (paulstelian97) [inline tools/logger/convert.c]: Maybe snprintf with a destination buffer size of 0 can be used to compute the size? Or is that usage nonstandard?

Comment (paulstelian97) [reply tools/logger/convert.c]: (even if 0 is nonstandard you can use size 1 and it still works)

Comment (paulstelian97) [reply tools/logger/convert.c]: Although possibly the best solution would be to provide an implementation of asprintf on systems which don't already have it. A simple one:

```
char *vasprintf(const char *format, va_list args)
{
	va_list args_copy;
	int size;
	char localbuf[1];
	char *result;

	va_copy(args_copy, args);
	size = vsnprintf(localbuf, 1, format, args_copy);
	va_end(args_copy);

	result = calloc(1, size + 1);
	if (result)
		vsnprintf(result, size + 1, format, args);
	return result;
}

char *asprintf(const char *format, ...)
{
	va_list args;
	char *result;

	va_start(args, format);
	result = vasprintf(format, args);
	va_end(args);

	return result;
}
```

Comment (ktrzcinx) [reply tools/logger/convert.c]: ok, introduced

File: tools/logger/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
```
Comment (dbaluta) [inline tools/logger/CMakeLists.txt]: Can you also put the warning in the commit message. It looks out of context.

Comment (ktrzcinx) [reply tools/logger/CMakeLists.txt]: Added.

> It looks out of context.

A little, but it's quite obvious problem and solution so I didn't create another PR for this, especially when we have some troubles with merging and number of PR grow. After merge there won't be any difference between two vs one PR) Moreover it's not any hotfix, so there is no need to introduce it very fast.

File: tools/logger/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
 
+cmake_minimum_required(VERSION 3.10)
```
Comment (jajanusz) [inline tools/logger/CMakeLists.txt]: Why did you add it? It's needed usually in top-level file (in this case tools).

Comment (jajanusz) [reply tools/logger/CMakeLists.txt]: Ok I see in commit message now. Anyway it ecnourages users to build logger as standalone tools (instead of being part of tools) IDK if we want to support both cases, better to manage it from one place.

Comment (ktrzcinx) [reply tools/logger/CMakeLists.txt]: I didn't notice that and use it as top level one. Deleted.

File: tools/logger/logger.c [hunk 4]
```diff
@@ -196,6 +198,9 @@ int main(int argc, char *argv[])
 			config.version_fw = 1;
 			config.version_file = optarg;
 			break;
+		case 'd':
```
Comment (lgirdwood) [inline tools/logger/logger.c]: Can you make sure all these options are listed in the help.

Comment (ktrzcinx) [reply tools/logger/logger.c]: I didn't notice any lack in help message:
flags: `ho:i:l:ps:c:u:tev:r`
```
Usage sof-logger <option(s)> <file(s)>
sof-logger:                             Display mailbox contents
sof-logger:      -i infile -o outfile   Dump infile contents to outfile
sof-logger:      -l *.ldc_file          *.ldc files generated by rimage
sof-logger:      -p                     Input from stdin
sof-logger:      -e                     Enable checking firmware version with default verification file
sof-logger:      -v ver_file            Enable checking firmware version with ver_file file
sof-logger:      -c                     Set timestamp clock in MHz
sof-logger:      -s state_name          Take a snapshot of state
sof-logger:      -t                     Display trace data
sof-logger:      -u baud                Input data from a UART
sof-logger:      -r                     Less formatted output for chained log processors
sof-logger:      -d                     Dump ldc information
```

Comment (paulstelian97) [reply tools/logger/logger.c]: The help for "c:" isn't exactly proper as flags expects an argument but that isn't mentioned in the help explicitly.

Comment (ktrzcinx) [reply tools/logger/logger.c]: added

**Comment by jajanusz** [comment]
Most of checkpatch complains are caused by the fact that it does not know that `*print` functions are like trace functions. But this one complain is legit:
```
WARNING: line over 80 characters
#22: FILE: tools/logger/logger.c:45:
+	fprintf(stdout, "%s:\t -c clock\t\tSet timestamp clock in MHz\n", APP_NAME);
```
Cos it doesn't end with literal, there is symbol.

**Comment by ktrzcinx** [comment]
@jajanusz fixed
